tool/net/redbean.c

```diff
@@ -4697,6 +4697,161 @@ static int LuaGetHttpReason(lua_State *L) {
   return 1;
 }
 
+static int EncodeJsonData(lua_State *L, char **buf, int level, char *numformat) {
+  size_t idx = -1;
+  size_t tbllen, buflen;
+  bool isarray;
+  int t = lua_type(L, idx);
+  if (level < 0) return luaL_argerror(L, 1, "too many nested tables");
+  if (LUA_TSTRING == t) {
+    appendw(buf, '"');
+    appends(buf, gc(EscapeJsStringLiteral(lua_tostring(L, idx), -1, 0)));
```

Comment:
[optional] if you want performance that's good to the last drop here:

```c
    tmpstr = EscapeJsStringLiteral(lua_tostring(L, idx), lua_string_len, &tmpstrlen);
    appendd(buf, tmpstr, tmpstrlen);
    free(tmpstr);
```

It's obviously less elegant than the thing you did before. I only make the suggestion since performance was mentioned. That's the best possible way you can get it without optimizing the underlying APIs.

Reply:
Right; I think it may be best to keep it the way it is, but it's a good suggestion to keep in mind.